lang/shared/src/main/scala/com/wavesplatform/lang/v1/compiler/Terms.scala

```diff
@@ -65,9 +67,33 @@ object Terms {
   sealed trait EVALUATED extends EXPR {
     def toStr: Coeval[String] = Coeval.now(toString)
   }
-  case class CONST_LONG(t: Long)        extends EVALUATED { override def toString: String = t.toString                 }
-  case class CONST_BYTESTR(bs: ByteStr) extends EVALUATED { override def toString: String = bs.toString                }
-  case class CONST_STRING(s: String)    extends EVALUATED { override def toString: String = s                          }
+  case class CONST_LONG(t: Long)        extends EVALUATED { override def toString: String = t.toString  }
+
+  case class CONST_BYTESTR private (bs: ByteStr) extends EVALUATED {
+    override def toString: String = bs.toString
+  }
+  object CONST_BYTESTR {
+    def apply(bs: ByteStr): Either[ExecutionError, CONST_BYTESTR] =
+      Either.cond(
+        bs.size <= DATA_TX_BYTES_MAX,
+        new CONST_BYTESTR(bs),
+        "ByteStr is too large"
+      )
+  }
+
+  case class CONST_STRING private (s: String) extends EVALUATED {
+    override def toString: String = s
+  }
+  object CONST_STRING {
+    def apply(s: String): Either[ExecutionError, CONST_STRING] =
+      Either.cond(
+        s.length * 2 <= DATA_TX_BYTES_MAX,
```

Comment:
are you sure length string should be capped by the value, not string's bytes length?